### PR TITLE
Improve srcflux documentation

### PR DESCRIPTION
**Pull request by hamogu**
This PR has two changes in one. Due to the way I wrote it, they are in the same commit,
but I can separate them if needed.

- The upper end of a credible interval is not the same as the upper limit for a non-detection.
  Previously, this was implied by the text, so I've added an explicit paragraph on that.
  For more details see, e.g. [Kashayap et al. (2010)](https://iopscience.iop.org/article/10.1088/0004-637X/719/1/900)
- aprates is a fully Basian computation, so the term "confidence interval" is not appropriate.

I have not run this through ahelp yet to check that it renders correctly, because I want to achieve agreement that we actually want to do these changes first.

Note: My editor is set to automatically remove trailing which space, so some lines appear to be edited without reason. I can revert those if you need me to, but I'd rather not do it now if I don't have to.

Note: While this also relates to srcflux, #520 is separate in that it only changes a single line in the output on screen for more pleasing formatting, while this PR includes substantial changes to the content and meaning of the documentation.

### CODE REVIEW

**Comment by DougBurke** [comment]
@hamogu - I split out the trailing-whitespace change to make the content change easier to read. It should be okay but the devil is in the rebase-details...

**Comment by DougBurke** [comment]
@anetasie - can you review Moritz's changes in commit https://github.com/cxcsds/ciao-contrib/pull/521/commits/76d36b1c9f3eae9da2f27fa8e5c2164263a96161 based on our script discussions earlier this week?

File: share/doc/xml/srcflux.xml [hunk 1]
```diff
@@ -69,11 +68,30 @@
...
 	which makes no correction, or psfmethod=psffile and provide
     an image of the surfrace brightness as the psffile parameter.
-        The other options are discussed below in the psfmethod parameter 
+        The other options are discussed below in the psfmethod parameter
      and the "PSF Correction" section.
       </PARA>
+      <PARA title='A fully Baysian computation'>
+	This tool is based on aprates, which performs a fully Baysian computation,
+	thus the result is not a (frequentist) confidence interval,
+	but should correctly be called credible interval.
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:77]: I would not get into the discussion of the difference between the correct/incorrect names, so confidence or credible interval in the documentation. Suggest change lines 76-77 to:
"thus the result is a credible interval (confidence interval for the frequentist calculation), which is CONF parameter."

File: share/doc/xml/srcflux.xml [hunk 2]
```diff
@@ -69,11 +68,30 @@
...
+	bound is reported. This number is sometimes erroneously used as the upper flux
+	limit for an undetected source, but strictly speaking the uncertainty on a flux
+	measurement and the upper limit of an undetected source are two different questions,
+	even though the numbers often come out to be in a similar range. When talking about the
+	upper limit for an undetected sources (the maximal flux that a source could have without
+	being detected), a single number for conf is not sufficient. Instead, two numbers are needed:
+	The probability of detecting a source in background fluctuations (false positive) and the
+	probability of missing a real source because it is compatible with background (false negative).
+	This calculation is beyond scope of this tool.
+      </PARA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:92]: Perhaps we should just quote the exact paragraph from Kashyap et al 2010 which has a very good description of the problem in Sec.2.1, or slightly shorter version, or just refer to this paragraph in the text. 

I would rephrase the current text and just state:
"The tool does not return upper limit and that the reported values are always calculated for lower and upper confidence bound. The upper limit calculation requires a detection threshold which does not dependent on the number of counts or the source flux. The detection threshold is not used in this tool. See Kashyap et al 2010, for full discussion of the difference between confidence bounds and upper limits."

File: share/doc/xml/srcflux.xml [hunk 3]
```diff
@@ -1005,30 +1024,30 @@ bounds(region(my.reg))
...
-      
+
       <PARAM name="conf" type="real" min="0" max="1" def="0.9" reqd="no">
 	<SYNOPSIS>Confidence interval</SYNOPSIS>
 	<DESC>
 	  <PARA>
-	    The confidence interval for the errors.  If the 
-	    net counts are 0, then the upper limit is reported.
+	    The credible interval for the errors. If the
+	    net counts are 0, then only the upper end of the range is reported.
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1046]: we can say "upper bound of the interval" instead of "end of the range"

File: share/doc/xml/srcflux.xml [hunk 4]
```diff
@@ -1595,33 +1614,33 @@ bounds(region(my.reg))
 	    NET_RATE, assuming non-informative priors for the
 	    intensities in the source and background apertures.  The
 	    mode of the PDF is the NET_RATE_APER.  A value of 0
-	    indicates that only an upper limit is available.</DATA>
+	    indicates that only the upper end of the credible interval is available.</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1617]: change to "the upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 5]
```diff
@@ -1595,33 +1614,33 @@ bounds(region(my.reg))
...
 	</ROW>
 	<ROW>
 	  <DATA>NET_RATE_APER_LO</DATA>
 	  &nodmx;
-	  <DATA>Lower equal tail confidence limit on NET_RATE.  A value
-	  of NaN indicates that only an upper limit is available as
+	  <DATA>Lower equal tail of the cradible interval on NET_RATE.  A value
+	  of NaN indicates that only the upper end of the credible interval is available as
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1623]: "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 6]
```diff
@@ -1595,33 +1614,33 @@ bounds(region(my.reg))
...
 	  the NET_RATE_APER_HI value. </DATA>
 	</ROW>
 	<ROW>
 	  <DATA>NET_RATE_APER_HI</DATA>
 	  &nodmx;
-	  <DATA>Upper equal tail confidence limit on NET_RATE.  If 
+	  <DATA>Upper equal tail of the credible interval on NET_RATE.  If
 	  the _LO value is NaN and the RATE is 0 then this value
-	  represent the upper limit on the count rate.</DATA>
+	  represent the upper end of the credible interval of the count rate.</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1631]: change to "upper bound of the credible interval for the count rate".

File: share/doc/xml/srcflux.xml [hunk 7]
```diff
@@ -1635,31 +1654,31 @@ bounds(region(my.reg))
...
 	<ROW>
 	  <DATA>NET_FLUX_APER_LO</DATA>
 	  &nodmx;
 	  <DATA>
-	    Lower confidence limit the model independent net flux computed
-	    by scaling the NET_RATE limits.  If only an upper limit
-	    was compute then it is used to scale the values.
+	    Lower bound of the credible interval for the model independent net flux computed
+	    by scaling the NET_RATE limits.  If only the upper end of the credible interval
+	    was computed then it is used to scale the values.
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1666]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 8]
```diff
@@ -1635,31 +1654,31 @@ bounds(region(my.reg))
...
 	  </DATA>
 	</ROW>
 	<ROW>
 	  <DATA>NET_FLUX_APER_HI</DATA>
 	  &nodmx;
 	  <DATA>
-	    Same as NET_FLUX_APER_LO for the upper confidence limit.
+	    Same as NET_FLUX_APER_LO for the upper end of the credible interval.
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1673]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 9]
```diff
@@ -1635,31 +1654,31 @@ bounds(region(my.reg))
...
 	  </DATA>
 	</ROW>
 	<ROW>
 	  <DATA>UPPER_LIMIT</DATA>
 	  &nodmx;
 	  <DATA>
-	    A flag identifying whether the  NET_FLUX_APER limits are 
-	    computed from an upper limit estimate (yes) or an observed NET_RATE (no).
+	    A flag identifying whether the  NET_FLUX_APER limits are
+	    computed from the upper end of the credible interval (yes) or an observed NET_RATE (no).
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1681]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 10]
```diff
@@ -1701,7 +1720,7 @@ bounds(region(my.reg))
     <ROW>
       <DATA>NET_PHOTFLUX_APER_HI</DATA>
       &nodmx;
-      <DATA>The upper confidence limit on the NET_PHOTFLUX_APER value obtained
+      <DATA>The upper end of the credible interval of on the NET_PHOTFLUX_APER value obtained
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1723]: change to "upper bound of the credible interval of the" and delete 'on'

File: share/doc/xml/srcflux.xml [hunk 11]
```diff
@@ -1725,12 +1744,12 @@ bounds(region(my.reg))
 	<ROW>
 	  <DATA>NET_MFLUX_APER_LO</DATA>
 	  &nodmx;
-	  <DATA>The lower confidence in NET_MFLUX_APER</DATA>
+	  <DATA>The lower end of the credible interval of NET_MFLUX_APER</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1747]: change to "lower band of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 12]
```diff
@@ -1725,12 +1744,12 @@ bounds(region(my.reg))
...
 	</ROW>
 	<ROW>
 	  <DATA>NET_MFLUX_APER_HI</DATA>
 	  &nodmx;
-	  <DATA>The upper confidence in the NET_MFLUX_APER</DATA>
+	  <DATA>The upper end of the credible interval in the NET_MFLUX_APER</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1752]: change to "upper bound of the credible interval of the"

File: share/doc/xml/srcflux.xml [hunk 13]
```diff
@@ -1742,19 +1761,19 @@ bounds(region(my.reg))
 	<ROW>
 	  <DATA>NET_UMFLUX_APER_LO</DATA>
 	  &nodmx;
-	  <DATA>Lower confidence limit on unabsorbed model flux</DATA>
+	  <DATA>Lower end of credible interval for unabsorbed model flux</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1764]: change to "lower bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 14]
```diff
@@ -1742,19 +1761,19 @@ bounds(region(my.reg))
...
 	</ROW>
 	<ROW>
 	  <DATA>NET_UMFLUX_APER_HI</DATA>
 	  &nodmx;
-	  <DATA>Upper confidence limit on the unabsorbed model flux</DATA>
+	  <DATA>Upper end of the credible interval for the unabsorbed model flux</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1769]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 15]
```diff
@@ -1837,12 +1856,12 @@ bounds(region(my.reg))
 </ROW>
 <ROW>
 <DATA>MERGED_NET_RATE_APER_LO</DATA>
-<DATA>Lower confidence level of merged net rate computed using
+<DATA>Lower end of the credible interval of merged net rate computed using
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1859]: change to "lower bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 16]
```diff
@@ -1837,12 +1856,12 @@ bounds(region(my.reg))
...
 aprates, the total counts, and exposure weighted PSFs</DATA>
 </ROW>
 <ROW>
 <DATA>MERGED_NET_RATE_APER_HI</DATA>
-<DATA>Same as previous for the upper confidence level.</DATA>
+<DATA>Same as previous for the upper end of the credible interval.</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1864]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 17]
```diff
@@ -1863,12 +1882,12 @@ images.
 <ROW>
 <DATA>MERGED_NET_APRATES_PHOTFLUX_APER_LO</DATA>
 <DATA>
-    The aprates_photflux lower confidence level 
+    The aprates_photflux lower end of credible interval
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1885]: change to "lower bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 18]
```diff
@@ -1863,12 +1882,12 @@ images.
...
 </DATA>
 </ROW>
 <ROW>
 <DATA>MERGED_NET_APRATES_PHOTFLUX_APER_HI</DATA>
-<DATA>    The aprates_photflux upper confidence level 
+<DATA>    The aprates_photflux upper end of credible interval
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1890]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 19]
```diff
@@ -1894,13 +1913,13 @@ The merged net photon flux by weighted averaging the per-observation values.
 </ROW>
 <ROW>
 <DATA>MERGED_NET_PHOTFLUX_APER_LO</DATA>
-<DATA>The lower confidence value on the net photon flux computed by scaling
+<DATA>The lower end of the credible inerval for the net photon flux computed by scaling
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1916]: change to "lower  bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 20]
```diff
@@ -1894,13 +1913,13 @@ The merged net photon flux by weighted averaging the per-observation values.
...
 the net rate errors.
 </DATA>
 </ROW>
 <ROW>
 <DATA>MERGED_NET_PHOTFLUX_APER_HI</DATA>
-<DATA>Same as above for the upper confidence value.
+<DATA>Same as above for the upper end of the credible interval.
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1922]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 21]
```diff
@@ -1922,26 +1941,26 @@ factor (MFLUX_CNV).
 </ROW>
 <ROW>
 <DATA>MERGED_NET_MFLUX_APER_LO</DATA>
-<DATA>Same as above for the lower confidence limit
+<DATA>Same as above for the lower end of credible interval
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1944]: change to "lower bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 22]
```diff
@@ -1922,26 +1941,26 @@ factor (MFLUX_CNV).
...
 </DATA>
 </ROW>
 <ROW>
 <DATA>MERGED_NET_MFLUX_APER_HI</DATA>
-<DATA>Same as above for the upper confidence limit.
+<DATA>Same as above for the upper end of the credible interval.
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1949]: change to "upper bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 23]
```diff
@@ -1922,26 +1941,26 @@ factor (MFLUX_CNV).
...
 <ROW>
 <DATA>MERGED_NET_UMFLUX_APER</DATA>
-<DATA>Same as the MERGED_NET_MFLUX_APER, but for just the unabsorbed 
+<DATA>Same as the MERGED_NET_MFLUX_APER, but for just the unabsorbed
 model component</DATA>
 </ROW>
 <ROW>
 <DATA>MERGED_NET_UMFLUX_APER_LO</DATA>
-<DATA>Same as above for the lower confidence limit</DATA>
+<DATA>Same as above for the lower end of credible interval</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1959]: change to "lower bound of the credible interval"

File: share/doc/xml/srcflux.xml [hunk 24]
```diff
@@ -1922,26 +1941,26 @@ factor (MFLUX_CNV).
...
 </ROW>
 <ROW>
 <DATA>MERGED_NET_UMFLUX_APER_HI</DATA>
-<DATA>Same as above for the upper confidence limit</DATA>
+<DATA>Same as above for the upper end of credible interval</DATA>
```
Comment (anetasie) [inline share/doc/xml/srcflux.xml:1963]: change to "upper bound of the credible interval"

**Review by anetasie — CHANGES_REQUESTED** [review]
@hamogu Thank you for these updates. I left some comments to the text and suggested changes.

**Review by anetasie — APPROVED** [review]
@hamogu Thanks for the update. It looks good.
